docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,663 @@
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_1,
+        provider
+    );
+    return wallet;
+}
+
+async function getWalletB() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_2,
+        provider
+    );
+    return wallet;
+}
+```
+
+### Creating our Lit Action and uploading it to IPFS
+
+Once we are done with our Lit Action, we need to upload it to IPFS so we get an immutable CID which will always point to our action.
+
+```js
+export async function createLitAction() {
+    console.log("creating lit action..");
+    const action = createERC20SwapLitAction(chainAParams, chainBParams);
+    const ipfsCid = await uploadViaPinata(action);
+
+    console.log("Lit Action code:\n", action);
+    console.log("IPFS CID: ", ipfsCid);
+    return ipfsCid;
+}
+```
+
+### Create a Mint/Grant/Burn PKP
+
+:::info
+Currently, direct calls to the smart contract’s m/g/b function are currently failing with the Lit Network. We’ll replicate the same functionality using the SDK.
+:::
+
+We'll create a single function to perform 3 calls at once
+
+- Mint a PKP with the user's wallet
+- Add the Lit Action as a permitted auth Method
+- Burn the PKP NFT by transferring it to itself so auth methods become immutable
+
+Now, our PKP will only be used to execute the Lit Action we permitted. It will generate signatures only when the Lit Action is running and not in any other situations. We have already designed the Lit Action to create the correct signed transactions based on the conditions.
+
+```js
+export async function mintGrantBurnPKP(action_ipfs, mintedPKP) {
+    console.log("minting started..");
+    const signerA = await getWalletA();
+
+// instantiating Contracts SDK
+    const litContracts = new LitContracts({
+        signer: signerA,
+        network: LitNetwork.DatilDev,
+        debug: false,
+    });
+    await litContracts.connect();
+
+// minting a new PKP
+    const mintPkp = await litContracts.pkpNftContractUtils.write.mint();
+    const pkp = mintPkp.pkp;
+    console.log("PKP: ", pkp);
+
+    console.log("adding permitted action..");
+
+// adding a permitted lit action
+    await litContracts.addPermittedAction({
+        pkpTokenId: pkp.tokenId,
+        ipfsId: action_ipfs,
+        authMethodScopes: [AuthMethodScope.SignAnything],
+    });
+
+    console.log("transfer started..");
+
+// burning the PKP NFT
+    const transferPkpOwnership =
+        await litContracts.pkpNftContract.write.transferFrom(
+            signerA.address,
+            pkp.ethAddress,
+            pkp.tokenId,
+            {
+                gasLimit: 125_000,
+            }
+        );
+
+    const receipt = await transferPkpOwnership.wait();
+
+    console.log(
+        "Transferred PKP ownership to itself: ",
+        receipt
+    );
+    return pkp;
+}
```

Comment:
This can be done in a single transaction using `litContracts.pkpHelperContract.write.mintNextAndAddAuthMethods` see [here](https://github.com/LIT-Protocol/custom-auth-telegram-example/blob/7aee88a01bc9feb42dd81f3f1cbfb17b5d0f470f/src/mintPkp.ts#L55-L71)

Reply:
I've earlier done this for simplicity and to improve readiness but would update it to use `mintNextAndAddAuthMethods ` now

Reply:
ah this slipped, will fix it in both code and docs